internal/connect/command_custom_plugin_delete.go

```diff
@@ -0,0 +1,68 @@
+package connect
+
+import (
+	"github.com/spf13/cobra"
+
+	pcmd "github.com/confluentinc/cli/v3/pkg/cmd"
+	"github.com/confluentinc/cli/v3/pkg/deletion"
+	"github.com/confluentinc/cli/v3/pkg/examples"
+	"github.com/confluentinc/cli/v3/pkg/resource"
+)
+
+func (c *customPluginCommand) newDeleteCommand() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "delete <id-1> [id-2] ... [id-n]",
+		Short: "Delete custom connector plugin.",
+		Args:  cobra.MinimumNArgs(1),
+		RunE:  c.delete,
+		Example: examples.BuildExampleString(
+			examples.Example{
+				Text: "Delete a custom connector plugin",
+				Code: "confluent connect custom-plugin delete ccp-123456",
+			},
+		),
+	}
+
+	pcmd.AddForceFlag(cmd)
+	pcmd.AddContextFlag(cmd, c.CLICommand)
+
+	return cmd
+}
+
+func (c *customPluginCommand) delete(cmd *cobra.Command, args []string) error {
+	pluginIdToName, err := c.mapPluginIdToName()
+	if err != nil {
+		return err
+	}
+
+	existenceFunc := func(id string) bool {
+		_, ok := pluginIdToName[id]
+		return ok
+	}
+
+	if err := deletion.ValidateAndConfirmDeletion(cmd, args, existenceFunc, resource.CustomConnectorPlugin, pluginIdToName[args[0]]); err != nil {
```

Comment:
This is the version which uses name validation for single arguments instead of a yes/no prompt, so only the first name is needed.

If multiple are provided, then the function won't actually use it.

We'll remove this and change everything to yes/no in v4.

Reply:
I think that is just the name of the plugin. My understanding is if there are more than 1 to delete then it would anyway convert to a yes or no question instead of typing the name it is only valid if there is 1 plugin to delete. I had based it on this https://github.com/confluentinc/cli/blob/main/internal/connect/command_cluster_delete.go#L60